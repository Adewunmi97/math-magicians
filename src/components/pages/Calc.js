import React from 'react';
import Calculator from '../Calculator';
import '../../App.css';

function Calc() {
  return (
    <div className="calc">
      <div className="calcontent">
        <h2 className="head-content"> Lets do some calculations</h2>
        <p className="content">
          Just type and your answer comes up live.This is a simple basic
          calculator with memory functions similar to a small handheld
          calculator. Use this basic calculator for math with addition,
          subtraction, division and multiplication. Memory and clear functions
          for this standard calculator are listed below.

          Numbers are important!. Whether costs, revenues, performance, targets â€“
          most people agree that numbers are important.
        </p>
      </div>
      <div>
        <Calculator />
      </div>
    </div>
  );
}

export default Calc;
