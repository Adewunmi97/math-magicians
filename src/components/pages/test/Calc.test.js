import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calc from '../Calc';

describe('test Calc component functionality', () => {
  it('should render Calc heading correctly', () => {
    const { container, getByRole } = render(<Calc />);
    expect(getByRole('heading')).toHaveTextContent('Lets do some calculations');
    expect(container).toMatchSnapshot();
  });
});
