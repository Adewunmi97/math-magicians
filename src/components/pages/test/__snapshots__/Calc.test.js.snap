// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Calc component functionality should render Calc heading correctly 1`] = `
<div>
  <div
    class="calc"
  >
    <div
      class="calcontent"
    >
      <h2
        class="head-content"
      >
         Lets do some calculations
      </h2>
      <p
        class="content"
      >
        Just type and your answer comes up live.This is a simple basic calculator with memory functions similar to a small handheld calculator. Use this basic calculator for math with addition, subtraction, division and multiplication. Memory and clear functions for this standard calculator are listed below. Numbers are important!. Whether costs, revenues, performance, targets – most people agree that numbers are important.
      </p>
    </div>
    <div>
      <div
        class="Calculator"
      >
        <div
          class="View"
          data-testid="screen"
        >
          <p />
          <p />
          <p>
            0
          </p>
        </div>
        <div
          class="UI"
        >
          <div
            class="group"
          >
            <button
              class="Button"
              type="button"
            >
              AC
            </button>
            <button
              class="Button"
              type="button"
            >
              +/-
            </button>
            <button
              class="Button"
              type="button"
            >
              %
            </button>
            <button
              class="Button ButtonOrange"
              type="button"
            >
              ÷
            </button>
          </div>
          <div
            class="group"
          >
            <button
              class="Button"
              type="button"
            >
              8
            </button>
            <button
              class="Button"
              type="button"
            >
              7
            </button>
            <button
              class="Button"
              type="button"
            >
              9
            </button>
            <button
              class="Button ButtonOrange"
              type="button"
            >
              x
            </button>
          </div>
          <div
            class="group"
          >
            <button
              class="Button"
              type="button"
            >
              4
            </button>
            <button
              class="Button"
              type="button"
            >
              5
            </button>
            <button
              class="Button"
              type="button"
            >
              6
            </button>
            <button
              class="Button ButtonOrange"
              type="button"
            >
              -
            </button>
          </div>
          <div
            class="group"
          >
            <button
              class="Button"
              type="button"
            >
              1
            </button>
            <button
              class="Button"
              type="button"
            >
              2
            </button>
            <button
              class="Button"
              type="button"
            >
              3
            </button>
            <button
              class="Button ButtonOrange"
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="group"
          >
            <button
              class="Button ButtonSpan2"
              type="button"
            >
              0
            </button>
            <button
              class="Button"
              type="button"
            >
              .
            </button>
            <button
              class="Button ButtonOrange"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
